### PR TITLE
fixed non-leading lecturer access to important areas

### PR DESCRIPTION
**Pull request by Ivanowicz**
Now lecturers signing into courses as students should have the same access as students, instead of being able to access and change what the main course lecturer could.

### CODE REVIEW

**Review by marcinz99 — APPROVED** [review]
Looks fine and seems to be working as intended

File: src/main/resources/templates/activity_lecturer.html [hunk 1]
```diff
@@ -141,15 +141,15 @@ <h6 class="align-top mb-3 border-bottom border-light mr-3 pb-3">Komentarz:</h6>
             </th>
         </tr>
         <thead style="border-top: 3px solid #343a40;">
-        <tr th:if="${#strings.contains(#strings.setJoin(user.getRoles(), ','), 'LECTURER')}">
+        <tr th:if="${#strings.contains(#strings.setJoin(user.getRoles(), ','), 'LECTURER') and user.getCoursesLectured().contains(course)}">
```
Comment (anosek-an) [inline src/main/resources/templates/activity_lecturer.html]: I don't think strings.contains(#strings.setJoin(user.getRoles(), ','), 'LECTURER') check is necessary - we initialize coursesLectured with an empty collection, so we don't need to worry about null pointer exceptions while checking with user.getCoursesLectured().contains(course). Thus, we can reduce this whole line to just a simple contains check.

File: src/main/java/upeldev/com/github/upel3/controllers/ActivityController.java [hunk 2]
```diff
@@ -69,11 +69,14 @@ public String activity(Model model, Principal principal, HttpServletRequest requ
...
 
+            Course course = activity.getCourse();
+            model.addAttribute("course", course);
+
             int passValue = activity.getPassValue();
             double maxPoints = activity.getMaxPoints();
             double points = 0;
             List<Grade> grades;
-            if(currentUser.getRoles().contains(Role.STUDENT)){
+            if(currentUser.getRoles().contains(Role.STUDENT) || (currentUser.getRoles().contains(Role.LECTURER) && !currentUser.getCoursesLectured().contains(course))){
```
Comment (anosek-an) [inline src/main/java/upeldev/com/github/upel3/controllers/ActivityController.java]: checking whether coursesEnrolledIn contains current course be better than those convoluted conditions

**Review by anosek-an — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/activity_lecturer.html (hunk 1 above)
Comment (Ivanowicz) [reply src/main/resources/templates/activity_lecturer.html]: Yup, good point, I just tacked it on the end without much thinking.

File: src/main/java/upeldev/com/github/upel3/controllers/ActivityController.java (hunk 2 above)
Comment (Ivanowicz) [reply src/main/java/upeldev/com/github/upel3/controllers/ActivityController.java]: Yeah, that's way simpler.

**Review by anosek-an — APPROVED** [review]
It's all right now :)

**Pull request merged**